lib/btree.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+const DEFAULT_DEGREE = 6; // min degree of b-tree.
+// All vertices except the root have [degree ... 2 * degree] child nodes
+// And [degree - 1 ... 2 * degree - 1] + 1 "empty" elements
+
+class Element {
+  constructor(value, tuple, less = null) {
+    this.value = value;
+    this.tuple = tuple;
+    this.less = less;
+  }
+}
+
+const empty = less => new Element(undefined, null, less);
+
+const destroyNode = (parent, index) => {
+  const node = parent[index].less;
+  const len = node.length;
+  const newLeftNode = node.splice(0, len / 2 - 1); // First half
+  const mid = node.shift();
+  newLeftNode.push(empty(mid.less));
+  mid.less = newLeftNode;
+  parent.splice(index, 0, mid);
+  return parent;
+};
+
+const isLeaf = node => !node[0].less;
```

Comment:
Is this correct? The element may have a subtree under any of the children AFAIK, isn't it? i.e.
```
[1, 5, 15, 30]
       ↙
     [7, 8]  
```

Reply:
In the b-tree, all elements from the node have children, or none. Therefore, we can only check the first element, and be sure that the rest of the elements in this node have a similar situation